### PR TITLE
chore: update dependencies and fix new lint/compiler issues

### PR DESCRIPTION
**Pull request by micmro**
To keep out build dependencies up to date.
Will need to wait for your type definition PR to add that in and add the `--strictNullChecks` flag to the build.

Also fixed the issues the lint and compiler flagged.

### CODE REVIEW

File: src/ts/transformers/har.ts [hunk 1]
```diff
@@ -242,9 +242,9 @@ const buildDetailTimingBlocks = (startRelative: number, harEntry: Entry): Waterf
     // special case for 'connect' && 'ssl' since they share time
     // http://www.softwareishard.com/blog/har-12-spec/#timings
     if (key === "connect" && t["ssl"] && t["ssl"] !== -1) {
-      const sslStart = parseInt(harEntry[`_ssl_start`], 10) || time.start;
-      const sslEnd = parseInt(harEntry[`_ssl_end`], 10) || time.start + t.ssl;
-      const connectStart = (!!parseInt(harEntry[`_ssl_start`], 10)) ? time.start : sslEnd;
+      const sslStart = parseInt(harEntry[`_ssl_start`].toString(), 10) || time.start;
```
Comment (tobli) [inline src/ts/transformers/har.ts]: This will fail if _ssl_start is missing.

Comment (micmro) [reply src/ts/transformers/har.ts]: Right, good catch, will wait for the ts definition merge, update it and fix all the adjacent issues that will pop up then.

**Comment by micmro** [comment]
Used @tobli's updated HAR type definition and enabled strict mode. Resulted in quite a few small changes. @tobli & @soulgalore can you guys have a quick look for regressions?

**Comment by micmro** [comment]
Released with `2.3.0`

**Pull request merged**